src/main/java/chattime/parser/Parser.java

```diff
@@ -0,0 +1,315 @@
+    private static final String NEED_INT = "OOPS!!! The index to %1$s must be positive integer.";
+    private static final String IDX_OUT_OF_BOUND = "OOPS!!! The index is too large! We currently have %d task(s).";
+    private static final String UNRECOGNISED_COMMAND = "Sorry... but I don't understand what you said >,<";
+
+    private static String userInput;
+    private static String[] splitCommand;
+    private static String command;
+    private static String description = null;
+
+    /**
+     * Returns suitable Command object for further execution.
+     *
+     * @param input Command and description entered by user.
+     * @return Command object to execute user request.
+     * @throws ChattimeException Exception thrown when command is not recognisable.
+     */
+    public static Command parse(String input) throws ChattimeException {
+        parseInput(input);
+        checkCleanCommand();
+
+        switch (command) {
+        case "bye":
+            return parseBye();
+
+        case "list":
+            return parseList();
+
+        case "todo":
+        case "deadline":
+        case "event":
+        case "listTime":
+        case "find":
+            switch (command) {
+            case "todo":
+                checkAddCommand();
+                return parseTodo();
+
+            case "deadline":
+                checkAddCommand();
+                return parseDeadline();
+
+            case "event":
+                checkAddCommand();
+                return parseEvent();
+
+            case "listTime":
+                return parseListTime();
+
+            case "find":
+                return parseFind();
+
+            default:
+                throw new ChattimeException(UNRECOGNISED_COMMAND);
+            }
+
+        case "mark":
+        case "unmark":
+        case "delete":
+
+            int index = checkIndexCommand();
+
+            if (command.equals("mark")) {
+                return new MarkCommand(index, true);
+
+            } else if (command.equals("unmark")) {
+                return new MarkCommand(index, false);
+
+            } else {
+                return new DeleteCommand(index);
+            }
+
+        default:
+            throw new ChattimeException(UNRECOGNISED_COMMAND);
+        }
+    }
+
+    /**
+     * Process user input into command and description.
+     *
+     * @param input Command and description entered by user.
+     */
+    private static void parseInput(String input) {
+        userInput = input;
+        splitCommand = userInput.split(" ", 2);
+        command = splitCommand[0];
+
+        if (splitCommand.length > 1) {
+            description = splitCommand[1];
+        }
+    }
+
+    /**
+     * Checks user's command clashes with data storage format.
+     *
+     * @throws ChattimeException Returns error message to require new input with problem statement.
+     */
+    private static void checkCleanCommand() throws ChattimeException {
```

Comment:
Very nice, checking for this edge case.

Reply:
Thank you for your comment !:star_struck: